Sources/HipMobileUI/ViewModels/NavigationViewModel.cs

```diff
@@ -52,6 +59,34 @@ public virtual void OnHidden()
         public virtual void OnRevealed()
         {
             AchievementNotification.ReloadDisplayedData();
+            StartAutoCheckForAchievements();
+        }
+
+        private int isAutoChecking = 0;
```

Comment:
Is there a particular reason why you are using an _int_ instead of a _bool_? I know it's technically almost the same but it would simplify the conditional statements.

Reply:
Yes, as noted in the PR description:

> `isAutoChecking` has to be an `int` because `Interlocked.Exchange` does not support `bool`